src/hooks/tests/usePatchStatusSelect.test.ts

```diff
@@ -0,0 +1,144 @@
+import { renderHook, act } from "@testing-library/react-hooks";
```

Comment:
I definitely agree e2e tests are extremely valuable as they allow you to test how the system operates as a whole and determine if the program works as intended. 

But i think its definitely still worth it and complimentary to write unit tests for complex functionality like this. Since you are able to ensure the function works as intended and its behavior is captured and defined by the test. When an e2e test breaks it might be difficult to determine why or how it broke but with more fine grained tests your able to determine if the features that build up to the e2e test are still behaving as expected and it makes it simple to debug why an e2e test is failing if you can determine what feature used in the test has broken or isnt behaving as intended.

Its also nice to be able to do TDD for something like this with unit tests. Since i can define the expected behavior and define edge cases for it And then build out and refine the functionality so i can capture the different scenarios this function may go through.

Reply:
I think it makes sense to unit test shared hooks because it's easier to rule out issues when e2e tests fail.  With the correct unit tests in place, it may not be necessary to select every combo of inputs in e2e tests to ensure a feature works as intended. What do ya'll think? @tgrander @khelif96

Reply:
e2e tests that are written well in cypress, meaning they are modular and test single units of the feature at a time, should not be difficult to debug. cypress makes debugging failed tests easy and personally i have not had trouble diagnosing a failing test unless the problem is with cypress itself, i.e. async difficulties. 

i am not going to write unit tests for most of my custom hooks because i plant to test them via the e2e tests. this hook specifically exists for one component. therefore it is essentially the component logic abstracted to a hook for better readability. since it is apart of the component i would test it thoroughly through the e2e tests.

i'm not going to object to anyone writing more extensive tests for a hook. you are correct that unit tests can catch more edge cases because they are not limited to a predefined set of data like the e2e tests are. however i will probably not write any unit tests...maybe ever. we'll see.